### PR TITLE
Feat: Implement ERC20 with Comprehensive Tests, Mock ERC20 Token

### LINKED ISSUE #126: [Feat][Lab Chef Needed] Implement ERC20, tests on ERC20 in contracts, and mock erc20
There's an erc721 folder in the `poker-texts-hold-em` folder, rename it to `erc` or `tokens`
In the src file there, write your `erc20.cairo` there, and write tests on it. 🥲

Then go to the game contracts, in the test file, then implement a mock `erc20.cairo`

I don't know if you understand.

Please use open zeppelin components, and do not repeat the interfaces implemented. You can create a new interface that has the `mint` function, and implement it, asserting the caller is the owner.

### PR DESCRIPTION
**Pull request by emarc99**
## Description   
    
This PR implements ERC20 OpenZeppelin standard. It has the following relevant test coverage:
- Initial State Verification
- ERC20 Functionality
- Ownable Features
- Edge Cases
- Upgradeability
## Related Issues   
    
- Closes: #126 
## How to Test  

`scarb test`    
 
## Screenshots (if applicable)  
    
image

  
## Checklist  
- [x] My code follows the project's coding style.  
- [x] I have tested these changes locally.

### CODE REVIEW

**Comment by emarc99** [comment]
@Birdmannn  kindly review. Will revert asap. Thanks!

File: poker-texas-hold-em/tokens/tests/test_erc20.cairo [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        assert(events.events.len() == 1, 'Upgrade event not emitted');
+        // Verify upgrade event
+    // let expected_upgrade_event =
+    //     UpgradeEvent::Upgraded(Upgraded { class_hash: new_class_hash })
+    // ;
+
+        // Assert that the event was emitted
+    // let expected_events = array![(upgradeable_dispatcher.contract_address,
+    // expected_upgrade_event)];
+    // spy.assert_emitted(@expected_events);
```
Comment (Birdmannn) [inline poker-texas-hold-em/tokens/tests/test_erc20.cairo]: Why did you comment out all the events sir? 👀

**Review by Birdmannn — COMMENTED** [review]
Nice

File: poker-texas-hold-em/tokens/tests/test_erc20.cairo (hunk 1 above)
Comment (emarc99) [reply poker-texas-hold-em/tokens/tests/test_erc20.cairo]: I'll fix it ser! 🥺😭

Comment (emarc99) [reply poker-texas-hold-em/tokens/tests/test_erc20.cairo]: Please, review once more kind ser. 🙏

**Review by Birdmannn — APPROVED** [review]
Trés bien, sir @emarc99 🎉 I'm very grateful for your help and expertise 🙇🏻💯. I hope we meet again ♠️
